pandas/_typing.py

```diff
@@ -84,9 +84,8 @@
 FrameOrSeries = TypeVar("FrameOrSeries", bound="NDFrame")
 
 Axis = Union[str, int]
-Label = Optional[Hashable]
-IndexLabel = Union[Label, Sequence[Label]]
-Level = Union[Label, int]
+IndexLabel = Union[Hashable, Sequence[Hashable]]
```

Comment:
great i would add some comments here on when to use IndexLabel and that Hashable is a thing. (can be a followup)

Reply:
maybe a sperate PR since we may want to discuss renaming to IndexLabelArg for consistency. not thought it through thoroughly yet though.

Reply:
main reason for this is to have it clear to the maintainer which annotations we should be using where, for consistency.